src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,463 @@
+        event.preventDefault();
+        event.stopPropagation();
+      },
+      [onChange, updateThumbValue],
+    );
+
+    const handlePointerDownOnSlider = React.useCallback(
+      (event: React.PointerEvent) => {
+        if (containerRef.current) {
+          const percent = getPercentageOfRectangle(
+            containerRef.current.getBoundingClientRect(),
+            event.pageX,
+          );
+          let pointerValue = min + (max - min) * percent;
+          pointerValue = roundValueToClosestStep(pointerValue, step, min);
+
+          const closestValueIndex = getClosestValueIndex(
+            currentValues,
+            pointerValue,
+          );
+          const [minVal, maxVal] = getAllowableThumbRange(closestValueIndex);
+          pointerValue = getBoundedValue(pointerValue, minVal, maxVal);
+          if (pointerValue === currentValues[closestValueIndex]) {
+            return;
+          }
+          const newValues = [...currentValues];
+          newValues[closestValueIndex] = pointerValue;
+          setCurrentValues(newValues);
+          onChange?.(newValues);
+        }
+      },
+      [min, max, step, currentValues, getAllowableThumbRange, onChange],
+    );
+
+    const pointerMoveFunctionRef = React.useRef<
+      (event: PointerEvent) => void
+    >();
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      if (pointerMoveFunctionRef.current && ownerDoc) {
+        ownerDoc.removeEventListener(
+          'pointermove',
+          pointerMoveFunctionRef.current,
+        );
+      }
+      pointerMoveFunctionRef.current = handlePointerMove;
+    }, [handlePointerMove]);
+
+    const pointerUpFunctionRef = React.useRef<(event: PointerEvent) => void>();
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      if (pointerUpFunctionRef.current && ownerDoc) {
+        ownerDoc.removeEventListener('pointerup', pointerUpFunctionRef.current);
+      }
+      pointerUpFunctionRef.current = handlePointerUp;
+    }, [handlePointerUp]);
+
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      ownerDoc?.addEventListener('pointermove', handlePointerMove);
+      ownerDoc?.addEventListener('pointerup', handlePointerUp);
+      return () => {
+        ownerDoc?.removeEventListener('pointermove', handlePointerMove);
+        ownerDoc?.removeEventListener('pointerup', handlePointerUp);
+      };
+    }, [handlePointerMove, handlePointerUp]);
+
+    return (
+      <div
+        className={cx(
+          'iui-slider-component-container',
+          disabled && 'iui-disabled',
+          className,
+        )}
+        {...rest}
+      >
+        {minValueLabel && (
+          <span className='iui-slider-min'>{minValueLabel}</span>
+        )}
+        <div
+          ref={refs}
+          className={cx(
+            'iui-slider-container',
+            undefined !== activeThumbIndex && 'iui-grabbing',
```

Comment:
This would be more readable as `{ 'iui-grabbing': !!activeThumbIndex }`.

I noticed you were doing this for `iui-active` in Thumb too. See [`STYLEGUIDE.md`](https://github.com/iTwin/iTwinUI-react/blob/main/STYLEGUIDE.md#use-classnames-object-syntax-for-conditional-classes).

Reply:
!!activeThumbIndex would result in false for `0` and `undefined`, correct? This would be wrong.

Reply:
Right, so it should be `{ 'iui-grabbing': activeThumbIndex !== undefined }`.

My "readability" point here really is about putting the classname first, since that's the thing we are toggling. Short circuiting was prohibited in our style guide for this reason.